assets/js/libs/dc-modal.js

```diff
@@ -0,0 +1,59 @@
+(function($){
+
+  window.Modal = {
+    NUMBER_OF_SECONDS_TO_DELAY_MODAL: 5,
+    DOM_ID: 'dc-modal',
+    DAYS_BETWEEN_MODAL_DISPLAYS: 5,
+    PROPORTION_OF_USERS_WHO_SHOULD_SEE_MODAL: 0.5,
+
+    openWithDelay: function() {
+      window.setTimeout(function() {
+        $('#' + window.Modal.DOM_ID).modal({
+          fadeDuration: 250,
+          fadeDelay: 0.8
+        });
+        window.Modal.setLocalStorageExpiration();
+      }, 1000 * this.NUMBER_OF_SECONDS_TO_DELAY_MODAL);
+    },
+
+    shouldRender: function() {
+      var endOfNoDisplayPeriod = localStorage.getItem('modal-expiration');
+      if(endOfNoDisplayPeriod) { return (new Date()).getTime() >= JSON.parse(endOfNoDisplayPeriod) }
```

Comment:
Let's say person `A` lands on the page. It gets assigned `'generic'` as `treatmentGroup`, and `'modal-expiration'` is set. After 5 days, they still have `'modal-expiration'` set right? That means they will always land in this part and the current date will always be more than 5 days after the modal expiration date.

Even if somehow 'modal-expiration' gets deleted once it's been more than 5 days, as long as they don't get assigned a new `treatmentGroup`, 'modal-expiration' won't be reset and they would end up in a situation where the modal would show on every occasion. 

Could be wrong somewhere btw, didn't have the chance to test the assumptions.

Reply:
Hmm, I think we reset the `modal-expiration` to 5 days from now whenever the modal opens:

https://github.com/datacamp/RDocumentation-app/pull/409/files#diff-b56e78f9098608cdeec3013b346e039bR15

So I think this should be fine, if I'm understanding correctly.

Reply:
Aha, I missed that, we're fine!